src/vector/sources/eventstoredb/cluster.rs

```diff
@@ -0,0 +1,163 @@
+            let mut gen = rand::rngs::SmallRng::from_entropy();
+
+            while ticks.next().await.is_some() {
+                match client.read_gossip().await {
+                    Err(error) => {
+                        tracing::error!(target: "eventstoredb_nexus_cluster_metrics", "{}", error)
+                    }
+                    Ok(members) => {
+                        let now = chrono::Utc::now();
+                        let tags = BTreeMap::new();
+                        let mut metrics = Vec::new();
+                        let dead_count = members.iter().filter(|m| !m.is_alive).count();
+                        let leaders = members
+                            .iter()
+                            .filter(|m| m.state == VNodeState::Leader)
+                            .collect::<Vec<&MemberInfo>>();
+
+                        if leaders.is_empty() {
+                            metrics.push(
+                                Metric::new(
+                                    "leader_mismatches",
+                                    MetricKind::Absolute,
+                                    MetricValue::Counter { value: 1.0 },
+                                )
+                                .with_namespace(Some(namespace.clone()))
+                                .with_tags(Some(tags.clone()))
+                                .with_timestamp(Some(now)),
+                            );
+                        }
+
+                        if dead_count > 0 {
+                            metrics.push(
+                                Metric::new(
+                                    "dead_nodes",
+                                    MetricKind::Absolute,
+                                    MetricValue::Counter {
+                                        value: dead_count as f64,
+                                    },
+                                )
+                                .with_namespace(Some(namespace.clone()))
+                                .with_tags(Some(tags.clone()))
+                                .with_timestamp(Some(now)),
+                            );
+                        }
+
+                        let current_epoch_number = if leaders.len() == 1 {
+                            leaders[0].epoch_number
+                        } else {
+                            let idx = gen.next_u32() % members.len() as u32;
+                            members[idx as usize].epoch_number
+                        };
+
+                        if epoch_number == i64::MAX {
+                            epoch_number = current_epoch_number;
+                        }
+
+                        if current_epoch_number != epoch_number {
+                            epoch_number = current_epoch_number;
+
+                            metrics.push(
+                                Metric::new(
+                                    "elections",
+                                    MetricKind::Absolute,
```

Comment:
Should this be Incremental? The idea is we'd add them up to ultimately see how many elections there were, right?

Reply:
Yeah, I was confused here by what the documentation said. I'll change it to `Incremental` then.